pkgs/development/compilers/openjdk/generic.nix

```diff
@@ -438,6 +409,12 @@ stdenv.mkDerivation (finalAttrs: {
   # still runs in parallel.
   enableParallelBuilding = false;
 
+  preConfigure =
+    # Set number of jobs to use when building.
+    ''
+      configureFlags+=("--with-jobs=''${NIX_BUILD_CORES}")
+    '';
+
```

Comment:
Previously, the build jobs were set based on the available cores on the machine and not the NIX_BUILD_CORES variable